dpnp/dpnp_iface_statistics.py

```diff
@@ -747,26 +803,95 @@ def var(x1, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     >>> import dpnp as np
     >>> a = np.array([[1, 2], [3, 4]])
     >>> np.var(a)
-    1.25
+    array(1.25)
+    >>> np.var(a, axis=0)
+    array([1.,  1.])
+    >>> np.var(a, axis=1)
+    array([0.25,  0.25])
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if x1_desc.size == 0:
-            pass
-        elif axis is not None:
-            pass
-        elif dtype is not None:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
+    if where is not True:
+        raise NotImplementedError(
+            "where keyword argument is only supported with its default value."
+        )
+    else:
+        if issubclass(a.dtype.type, dpnp.complexfloating):
+            # Note that if dtype is not of inexact type then arrmean will not be either.
+            arrmean = dpnp.mean(
+                a, axis=axis, dtype=dtype, keepdims=True, where=where
+            )
+            x = dpnp.subtract(a, arrmean, where=where)
+            x = dpnp.multiply(x, x.conj(), where=where).real
+            res = dpnp.sum(
+                x, axis=axis, dtype=dtype, keepdims=keepdims, where=where
+            )
+            if dtype is None and a.real.dtype != res.dtype:
+                res = res.astype(a.real.dtype, casting="same_kind")
+
+            cnt = _count_reduce_items(a, axis, where)
+            dof = dpnp.max(
+                dpnp.array(
+                    [cnt - ddof, 0],
+                    dtype=res.dtype,
+                    sycl_queue=res.sycl_queue,
+                    device=res.device,
+                )
+            )
+            if not dof:
+                dof = dpnp.nan
```

Comment:
the result looks different with numpy:
```python
a = numpy.arange(20, dtype='c8').reshape((4, 5))
a.var(axis=1, ddof=5)
# Out: array([inf, inf, inf, inf], dtype=float32)

b = dpnp.array(a)
b.var(axis=1, ddof=5)
# Out: array([nan, nan, nan, nan], dtype=float32)
```
It looks dpctl results the same, but not aligned with numpy. Is it expected or an issue which needs to be addressed to dpctl?

Reply:
array api indicates that for this special case, output should be `NaN`.

https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.var.html

Reply:
no it's clear, thank you